# <span style="font-size: 20pt; font-family: Montserrat, sans-serif; color: rgb(92, 148, 206); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Graded Assignment 3: 9 to 5</span>

  

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to show off your SQL skills! For each question, copy the SQL query you used and make note of the answer.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For this assignment, you will be using the Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results which can be found on</span> [<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(17, 85, 204); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; text-decoration-line: underline; vertical-align: baseline; white-space: pre-wrap;">Kaggle</span>](https://www.kaggle.com/bls/employment)<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Business Issue</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">You are working for the Bureau of Labor Statistics with the United States government and have been approached by your boss with an important meeting request. You have been asked by your supervisor to meet with Dolly Parton whose nonprofit is looking to shed light on the state of employment in the United States. As part of the 9 to 5 project, their research is focused on production and nonsupervisory employees and how those employees fare compared to all employees in the United States. While the data the BLS collects from the CES is publicly available, Dolly Parton and her colleagues need your assistance navigating the thousands of rows in each table in LaborStatisticsDB.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">About the Dataset</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">This dataset comes directly from the Bureau of Labor Statistics’ Current Employment Survey (CES). Here are some things you need to know:</span>

1. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">The industry table contains an NAICS code. This is different from the industry code. NAICS stands for North American Industry Classification System.</span>
    
2. <span style="font-size: 10pt; background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Series ID is composed of multiple different codes. CES stands for Current Employment Survey, the name of the survey which collected the data. The industry code as specified by the BLS and the data type code as specified in the datatype table.</span>
    

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Set Up</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To connect to the database, use the same connection info used during the SQL lessons.&nbsp;</span> 

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">For the assignment, we will be using the LaborStatisticsDB database.</span>

  

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Database Exploration</span>

<span style="font-size: 10pt; font-family: &quot;Times New Roman&quot;; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">To start with, let’s get to know the database further.</span>

1. <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: &quot;Times New Roman&quot;; font-size: 10pt; white-space: pre-wrap;">Use this space to make note of each table in the database, the columns within each table, each column’s data type, and how the tables are connected. You can write this down or draw a diagram. Whatever method helps you get an understanding of what is going on with LaborStatisticsDB.</span>

[erd](https:\github.com\KavVij\Assignment3_9to5_Kav\blob\main\erdjpg.jpg)

  

To add a photo, diagram or document to your file, drop the file into the folder that holds this notebook.  Use the the link button to the right of the  \</\> symbol in the gray part of this cell, the link is just the name of your file.

2. <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">What is the datatype for women employees? <b>Data type code 10</b></span>

In [138]:

select top 10 d.data_type_code
from LaborStatisticsDB.dbo.datatype d
join LaborStatisticsDB.dbo.series  s
on s.data_type_code=d.data_type_code
where d.data_type_text like '%women%'


(10 rows affected)

Total execution time: 00:00:00.709

data_type_code 10 10 10 10 10 10 10 10 10 10

3. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">What is the series id for&nbsp; women employees in the commercial banking industry in the financial activities supersector? </span> <span style="font-size: 13.3333px; white-space: pre-wrap;"><b>CES5552211010, CEU5552211010</b></span>

In [25]:

select series_id ,i.industry_code,industry_name,supersector_name
from LaborStatisticsDB.dbo.series s
Join LaborStatisticsDB.dbo.supersector as su
on s.supersector_code=su.supersector_code
join LaborStatisticsDB.dbo.industry as i
on s.industry_code= i.industry_code

where series_title like'%women%' and
      industry_name like'%commercial banking%' and
         supersector_name like '%financial activities%'



(2 rows affected)

Total execution time: 00:00:00.343

series_id industry_code industry_name supersector_name CES5552211010 55522110 Commercial banking Financial activities CEU5552211010 55522110 Commercial banking Financial activities

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Join in on the Fun</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Time to start joining! You can choose the type of join you use, just make sure to make a&nbsp; note!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join&nbsp; annual_2016 with series on series_id. We only want the data in the annual_2016 table to be included in the result.</span>

In [32]:
-- As they want only the data from annual_2016  am using exists as its more efficient. Marking in comments below alternate way for achieving the same
select *
from LaborStatisticsDB.dbo.annual_2016 a
where  exists (select * from LaborStatisticsDB.dbo.series s
where  a.series_id=s.series_id)

/*

select a.*
from LaborStatisticsDB.dbo.annual_2016 as a 
join LaborStatisticsDB.dbo.series as s 
on a.series_id=s.series_id
*/

(29042 rows affected)

Displaying Top 5000 rows.

Total execution time: 00:00:13.233

id series_id year period value footnote_codes original_file 0 CEU5500000007 2016 M13 36.9 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 1 CEU5500000008 2016 M13 26.11 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 2 CEU5500000030 2016 M13 962.73 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 3 CEU5500000031 2016 M13 411.29 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 4 CEU5500000032 2016 M13 11.15 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 5 CEU5500000034 2016 M13 111.6 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 6 CEU5500000035 2016 M13 179.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 7 CEU5500000081 2016 M13 236997 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 8 CEU5500000082 2016 M13 6189003 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 9 CEU5552200007 2016 M13 37.3 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 10 CEU5552200008 2016 M13 22.37 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 11 CEU5552200030 2016 M13 833.43 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 12 CEU5552200031 2016 M13 356.05 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 13 CEU5552200032 2016 M13 9.56 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 14 CEU5552200034 2016 M13 104.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 15 CEU5552200035 2016 M13 162.5 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 16 CEU5552200081 2016 M13 73459 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 17 CEU5552200082 2016 M13 1643184 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 18 CEU5552210007 2016 M13 37 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 19 CEU5552210008 2016 M13 20.25 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 20 CEU5552210030 2016 M13 749.49 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 21 CEU5552210031 2016 M13 320.19 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 22 CEU5552210032 2016 M13 8.65 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 23 CEU5552210034 2016 M13 101.8 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 24 CEU5552210035 2016 M13 161.7 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 25 CEU5552210081 2016 M13 46102 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 26 CEU5552210082 2016 M13 933434 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 27 CEU5552211007 2016 M13 37.1 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 28 CEU5552211008 2016 M13 20.35 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 29 CEU5552211030 2016 M13 754.09 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 30 CEU5552211031 2016 M13 322.16 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 31 CEU5552211032 2016 M13 8.69 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 32 CEU5552211034 2016 M13 106.8 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 33 CEU5552211035 2016 M13 173.2 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 34 CEU5552211081 2016 M13 35082 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 35 CEU5552211082 2016 M13 713870 NULL ce.data.55c.FinancialActivities.ProductionEmployeeHoursAndEarnings.csv 36 CEU5552212007 2016 M13 36.4 NULL ce.data.55c.Financia

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and datatype on data_type_code</span>

In [1]:
select top 50 *
from LaborStatisticsDB.dbo.series as s
join LaborStatisticsDB.dbo.datatype d 
on s.data_type_code=s.data_type_code

(50 rows affected)

Total execution time: 00:00:00.194

series_id supersector_code industry_code data_type_code seasonal series_title data_type_code data_type_text CES0000000001 0 00000000 1 S All employees 1 ALL EMPLOYEES CES0000000010 0 00000000 10 S Women employees 1 ALL EMPLOYEES CES0000000025 0 00000000 25 S All employees 1 ALL EMPLOYEES CES0000000026 0 00000000 26 S All employees 1 ALL EMPLOYEES CES0500000001 5 05000000 1 S All employees 1 ALL EMPLOYEES CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 ALL EMPLOYEES CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 ALL EMPLOYEES CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000010 5 05000000 10 S Women employees 1 ALL EMPLOYEES CES0500000011 5 05000000 11 S Average weekly earnings of all employees 1 ALL EMPLOYEES CES0500000012 5 05000000 12 S Average weekly earnings of all employees 1 ALL EMPLOYEES CES0500000013 5 05000000 13 S Average hourly earnings of all employees 1 ALL EMPLOYEES CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 1 ALL EMPLOYEES CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 1 ALL EMPLOYEES CES0500000019 5 05000000 19 S Average weekly hours of all employees 1 ALL EMPLOYEES CES0500000021 5 05000000 21 S Diffusion indexes 1 ALL EMPLOYEES CES0500000022 5 05000000 22 S Diffusion indexes 1 ALL EMPLOYEES CES0500000023 5 05000000 23 S Diffusion indexes 1 ALL EMPLOYEES CES0500000025 5 05000000 25 S All employees 1 ALL EMPLOYEES CES0500000026 5 05000000 26 S All employees 1 ALL EMPLOYEES CES0500000030 5 05000000 30 S Average weekly earnings of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000031 5 05000000 31 S Average weekly earnings of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000032 5 05000000 32 S Average hourly earnings of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000034 5 05000000 34 S Indexes of aggregate weekly hours of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000035 5 05000000 35 S Indexes of aggregate weekly payrolls of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000036 5 05000000 36 S Average weekly hours 1 ALL EMPLOYEES CES0500000056 5 05000000 56 S Aggregate weekly hours of all employees 1 ALL EMPLOYEES CES0500000057 5 05000000 57 S Aggregate weekly payrolls of all employees 1 ALL EMPLOYEES CES0500000081 5 05000000 81 S Aggregate weekly hours of production and nonsupervisory employees 1 ALL EMPLOYEES CES0500000082 5 05000000 82 S Aggregate weekly payrolls of production and nonsupervisory employees 1 ALL EMPLOYEES CES0600000001 6 06000000 1 S All employees 1 ALL EMPLOYEES CES0600000002 6 06000000 2 S Average weekly hours of all employees 1 ALL EMPLOYEES CES0600000003 6 06000000 3 S Average hourly earnings of all employees 1 ALL EMPLOYEES CES0600000006 6 06000000 6 S Production and nonsupervisory employees 1 ALL EMPLOYEES CES0600000007 6 06000000 7 S Average weekly hours of production and nonsupervisory employees 1 ALL EMPLOYEES CES0600000008 6 06000000 8 S Average hourly earnings of production and nonsupervisory employees 1 ALL EMPLOYEES CES0600000010 6 06000000 10 S Women employees 1 ALL EMPLOYEES CES0600000011 6 06000000 11 S Average weekly earnings of all employees 1 ALL EMPLOYEES CES0600000012 6 06000000 12 S Average weekly earnings of all employees 1 ALL EMPLOYEES CES0600000013 6 06000000 13 S Average hourly earnings of all employees 1 ALL EMPLOYEES CES0600000016 6 06000000 16 S Indexes of aggregate weekly hours of all employees 1 ALL EMPLOYEES CES0600000017 6 06000000 17 S Indexes of aggregate weekly payrolls of all employees 1 ALL EMPLOYEES CES0600000025 6 06000000 25 S All employees 1 ALL EMPLOYEES CES0600000030 6 06000000 30 S 

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Join series and industry on industry_code</span>

In [2]:
select top 50  *
from LaborStatisticsDB.dbo.series s 
join LaborStatisticsDB.dbo.industry as i 
on s.industry_code=i.industry_code


(50 rows affected)

Total execution time: 00:00:00.141

series_id supersector_code industry_code data_type_code seasonal series_title id industry_code naics_code publishing_status industry_name display_level selectable sort_sequence CES0000000001 0 00000000 1 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000010 0 00000000 10 S Women employees 0 0 - B Total nonfarm 0 T 1 CES0000000025 0 00000000 25 S All employees 0 0 - B Total nonfarm 0 T 1 CES0000000026 0 00000000 26 S All employees 0 0 - B Total nonfarm 0 T 1 CES0500000001 5 05000000 1 S All employees 1 5000000 - A Total private 1 T 2 CES0500000002 5 05000000 2 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000003 5 05000000 3 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000006 5 05000000 6 S Production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000007 5 05000000 7 S Average weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000008 5 05000000 8 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000010 5 05000000 10 S Women employees 1 5000000 - A Total private 1 T 2 CES0500000011 5 05000000 11 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000012 5 05000000 12 S Average weekly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000013 5 05000000 13 S Average hourly earnings of all employees 1 5000000 - A Total private 1 T 2 CES0500000016 5 05000000 16 S Indexes of aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000017 5 05000000 17 S Indexes of aggregate weekly payrolls of all employees 1 5000000 - A Total private 1 T 2 CES0500000019 5 05000000 19 S Average weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000021 5 05000000 21 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000022 5 05000000 22 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000023 5 05000000 23 S Diffusion indexes 1 5000000 - A Total private 1 T 2 CES0500000025 5 05000000 25 S All employees 1 5000000 - A Total private 1 T 2 CES0500000026 5 05000000 26 S All employees 1 5000000 - A Total private 1 T 2 CES0500000030 5 05000000 30 S Average weekly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000031 5 05000000 31 S Average weekly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000032 5 05000000 32 S Average hourly earnings of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000034 5 05000000 34 S Indexes of aggregate weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000035 5 05000000 35 S Indexes of aggregate weekly payrolls of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000036 5 05000000 36 S Average weekly hours 1 5000000 - A Total private 1 T 2 CES0500000056 5 05000000 56 S Aggregate weekly hours of all employees 1 5000000 - A Total private 1 T 2 CES0500000057 5 05000000 57 S Aggregate weekly payrolls of all employees 1 5000000 - A Total private 1 T 2 CES0500000081 5 05000000 81 S Aggregate weekly hours of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0500000082 5 05000000 82 S Aggregate weekly payrolls of production and nonsupervisory employees 1 5000000 - A Total private 1 T 2 CES0600000001 6 06000000 1 S All employees 2 6000000 - A Goods-producing 1 T 3 CES0600000002 6 06000000 2 S Average weekly hours of all employees 2 6000000 - A Goods-producing 1 T 3 CES0600000003 6 06000000 3 S Average hourly earnings of all employees 2 6000000 - A Goods-producing 1 T 3 CES0600000006 6 06000000 6 S Production and nonsupervisory employees 2 6000000 - A Goods-producing 1 T 3 CES0600000007 6 06000000 7 S Average weekly hours of production and nonsupervisory employees 2 6000000 - A Goods-producing 1 T 3 C

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Aggregate Your Friends and Code some SQL</span>

<span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Put together the following:</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many employees were reported in 2016 in all industries? Round to the nearest whole number.&nbsp; No chopping people into little bits, please.<b> Ans:</b></span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>2340612</b></span>

In [130]:
select round(sum (value),0) headcount,series_title
from LaborStatisticsDB.dbo.annual_2016 as a ,LaborStatisticsDB.dbo.series as s 
where a.series_id=s.series_id
group by series_title
having series_title like lower('All employees')




(1 row affected)

Total execution time: 00:00:00.130

headcount series_title 2340612 All employees

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many women employees were reported in 2016 in all industries? Round to the nearest whole number.<b>&nbsp;Ans.</b></span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>1125490</b></span>

In [136]:
select round(sum (value),0) count,series_title
from LaborStatisticsDB.dbo.annual_2016 as a ,LaborStatisticsDB.dbo.series as s 
where a.series_id=s.series_id
AND series_title like lower('women employees')
group by series_title

(1 row affected)

Total execution time: 00:00:00.083

count series_title 1125490 Women employees

3. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">How many production/nonsupervisory employees were reported in 2016? Round to the nearest whole number.&nbsp;<b>Ans.</b></span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>1263650</b></span>

In [131]:
select round(sum (value),0) headcount,series_title
from LaborStatisticsDB.dbo.annual_2016 as a ,LaborStatisticsDB.dbo.series as s 
where a.series_id=s.series_id
and series_title = 'Production and nonsupervisory employees'
group by series_title

(1 row affected)

Total execution time: 00:00:00.077

headcount series_title 1263650 Production and nonsupervisory employees

4. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2017, what is the average weekly hours worked by production and nonsupervisory employees across all industries? <b>Ans.</b></span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>36.06</b></span>

In [108]:
/*select count (j.series_id) headcount,series_title
from LaborStatisticsDB.dbo.january_2017 as j ,LaborStatisticsDB.dbo.series as s 
where j.series_id=s.series_id
group by series_title
having series_title like lower('Production and nonsupervisory employees')
*/

select round(avg(value),2)
from LaborStatisticsDB.dbo.january_2017 as j ,LaborStatisticsDB.dbo.series as s 
where j.series_id=s.series_id
      and lower(series_title)='average weekly hours of production and nonsupervisory employees'


(1 row affected)

Total execution time: 00:00:00.103

(No column name) 36.06

5. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">What is the total weekly payroll for production and nonsupervisory employees across all industries in January 2017? Round to the nearest penny, please. <b>Ans.</b></span><span style="font-size: 13.3333px; white-space: pre-wrap;"><b>1838753220</b></span>

In [123]:

select  sum(value) total
from LaborStatisticsDB.dbo.series as s,LaborStatisticsDB.dbo.january_2017 j
where s.series_id=j.series_id and 
     lower(series_title) like  ('aggregate weekly payrolls % production and nonsupervisory employees')
    




(1 row affected)

Total execution time: 00:00:00.093

total 1838753220

6. <span style="background-color: rgb(255, 255, 255);"><font color="#666666" face="Arvo, serif"><span style="font-size: 10pt; white-space: pre-wrap;">In January 2017, which industry was the average weekly hours worked by production and nonsupervisory employees the highest? Which industry was the lowest? <b>Ans.</b></span></font></span>Motor vehicle power train components <span style="color: rgb(102, 102, 102); font-family: Arvo, serif; background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap;"> is the HIGHEST and </span> <span style="color: rgb(102, 102, 102); font-family: Arvo, serif; background-color: rgb(255, 255, 255); font-size: 13.3333px; white-space: pre-wrap;">Fitness and recreational sports centers is the LOWEST .</span>

In [132]:
select distinct industry_name ,j.value
from LaborStatisticsDB.dbo.january_2017 j ,LaborStatisticsDB.dbo.industry as i,LaborStatisticsDB.dbo.series as s 
where s.series_id=j.series_id and 
      s.industry_code=i.industry_code and 
      i.industry_name is not null and
     lower(s.series_title) ='average weekly hours of production and nonsupervisory employees'
and j.value = (
select max(j17.value)
from LaborStatisticsDB.dbo.january_2017 j17 ,LaborStatisticsDB.dbo.industry as i17,LaborStatisticsDB.dbo.series as s17
where s17.series_id=j17.series_id and 
      s17.industry_code=i17.industry_code and 
      i17.industry_name is not null and
     lower(s17.series_title) ='average weekly hours of production and nonsupervisory employees'
)
order by industry_name


   

(1 row affected)

Total execution time: 00:00:00.107

industry_name value Motor vehicle power train components 49.8

In [133]:
select distinct industry_name ,j.value
from LaborStatisticsDB.dbo.january_2017 j ,LaborStatisticsDB.dbo.industry as i,LaborStatisticsDB.dbo.series as s 
where s.series_id=j.series_id and 
      s.industry_code=i.industry_code and 
      i.industry_name is not null and
     lower(s.series_title) ='average weekly hours of production and nonsupervisory employees'
and j.value = (
select min(j17.value)
from LaborStatisticsDB.dbo.january_2017 j17 ,LaborStatisticsDB.dbo.industry as i17,LaborStatisticsDB.dbo.series as s17
where s17.series_id=j17.series_id and 
      s17.industry_code=i17.industry_code and 
      i17.industry_name is not null and
     lower(s17.series_title) ='average weekly hours of production and nonsupervisory employees'
)
order by industry_name



(1 row affected)

Total execution time: 00:00:00.287

industry_name value Fitness and recreational sports centers 16.7

7. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In January 2021, which industry was the total weekly payroll for production and nonsupervisory employees the highest? Which industry was the lowest? <b>Ans</b>.Total Private has HIGHEST and </span>  <span style="font-size: 13.3333px; white-space: pre-wrap;">Coin-operated laundries and drycleaners&nbsp;has the LOWEST.</span>

In [142]:
select distinct industry_name ,j.value,s.series_title
from LaborStatisticsDB.dbo.january_2017 j ,LaborStatisticsDB.dbo.industry as i,LaborStatisticsDB.dbo.series as s 
where s.series_id=j.series_id and 
      s.industry_code=i.industry_code and 
      i.industry_name is not null and
     lower(s.series_title) like '%Aggregate weekly payrolls of production and nonsupervisory employees'
and j.value in (
select max(j17.value)
from LaborStatisticsDB.dbo.january_2017 j17 ,LaborStatisticsDB.dbo.industry as i17,LaborStatisticsDB.dbo.series as s17
where s17.series_id=j17.series_id and 
      s17.industry_code=i17.industry_code and 
      i17.industry_name is not null and
     lower(s17.series_title) like '%Aggregate weekly payrolls of production and nonsupervisory employees%'
)
order by industry_name

(1 row affected)

Total execution time: 00:00:00.448

industry_name value series_title Total private 74498171 Aggregate weekly payrolls of production and nonsupervisory employees

In [143]:
select distinct industry_name ,j.value,s.series_title
from LaborStatisticsDB.dbo.january_2017 j ,LaborStatisticsDB.dbo.industry as i,LaborStatisticsDB.dbo.series as s 
where s.series_id=j.series_id and 
      s.industry_code=i.industry_code and 
      i.industry_name is not null and
     lower(s.series_title) = 'Aggregate weekly payrolls of production and nonsupervisory employees'
and j.value in (
select min(j17.value)
from LaborStatisticsDB.dbo.january_2017 j17 ,LaborStatisticsDB.dbo.industry as i17,LaborStatisticsDB.dbo.series as s17
where s17.series_id=j17.series_id and 
      s17.industry_code=i17.industry_code and 
      i17.industry_name is not null and
     lower(s17.series_title) = 'Aggregate weekly payrolls of production and nonsupervisory employees'
)
order by industry_name

(1 row affected)

Total execution time: 00:00:00.243

industry_name value series_title Coin-operated laundries and drycleaners 10079 Aggregate weekly payrolls of production and nonsupervisory employees

## <span style="font-size: 12pt; font-family: Montserrat, sans-serif; color: rgb(53, 74, 95); background-color: rgb(255, 255, 255); font-weight: 700; font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Subqueries, Unions, Derived Tables, Oh My!</span>

1. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Write a query that returns the series_id, industry_code, industry_name, and value from the january_2017 table but only if that value is greater than the average value for annual_2016 of data_type_code 82.</span>

In [96]:
select distinct j.series_id,i.industry_code,industry_name,j.value
from LaborStatisticsDB.dbo.january_2017 j,LaborStatisticsDB.dbo.industry i,LaborStatisticsDB.dbo.series s
where j.series_id=s.series_id AND 
      i.industry_code=s.industry_code and 
            j.value> (
                      select avg(a.value)
                      from LaborStatisticsDB.dbo.annual_2016 a,LaborStatisticsDB.dbo.series s ,LaborStatisticsDB.dbo.datatype d
                    where d.data_type_code ='82'
                 and s.series_id =a.series_id 
                 and d.data_type_code=s.data_type_code)
                 



(377 rows affected)

Total execution time: 00:00:00.258

series_id industry_code industry_name value CES0500000056 5000000 Total private 4239112 CES0500000057 5000000 Total private 110301694 CES0500000081 5000000 Total private 3412651 CES0500000082 5000000 Total private 74498171 CES0600000057 6000000 Goods-producing 21779094 CES0600000082 6000000 Goods-producing 13426820 CES0800000056 8000000 Private service-providing 3432382 CES0800000057 8000000 Private service-providing 88280865 CES0800000081 8000000 Private service-providing 2818821 CES0800000082 8000000 Private service-providing 60914722 CES1000000057 10000000 Mining and logging 966707 CES1021000057 10210000 Mining 927250 CES2000000057 20000000 Construction 7555007 CES2000000082 20000000 Construction 5335274 CES2023600057 20236000 Construction of buildings 1765016 CES2023600082 20236000 Construction of buildings 1008035 CES2023620057 20236200 Nonresidential building 990555 CES2023700057 20237000 Heavy and civil engineering construction 1221764 CES2023700082 20237000 Heavy and civil engineering construction 908374 CES2023800057 20238000 Specialty trade contractors 4595075 CES2023800082 20238000 Specialty trade contractors 3421902 CES2023810057 20238100 Building foundation and exterior contractors 833625 CES2023820057 20238200 Building equipment contractors 2326342 CES2023820082 20238200 Building equipment contractors 1696080 CES2023821057 20238210 Electrical contractors 1039534 CES2023822057 20238220 Plumbing and HVAC contractors 1115270 CES3000000057 30000000 Manufacturing 13282613 CES3000000082 30000000 Manufacturing 7491723 CES3100000057 31000000 Durable goods 8769944 CES3100000082 31000000 Durable goods 4859716 CES3133200057 31332000 Fabricated metal products 1421283 CES3133200082 31332000 Fabricated metal products 871298 CES3133300057 31333000 Machinery 1251217 CES3133400057 31334000 Computer and electronic products 1489440 CES3133600057 31336000 Transportation equipment 2130760 CES3133600082 31336000 Transportation equipment 1257764 CES3133600157 31336001 NULL 1050677 CES3133640057 31336400 Aerospace products and parts 879785 CES3200000057 32000000 Nondurable goods 4496035 CES3200000082 32000000 Nondurable goods 2633628 CES3231100057 32311000 Food manufacturing 1293730 CES3231100082 32311000 Food manufacturing 851406 CES3232500057 32325000 Chemicals 1083956 CES4000000056 40000000 Trade 939889 CES4000000057 40000000 Trade 21232093 CES4000000082 40000000 Trade 14850661 CES4142000057 41420000 Wholesale trade 6858602 CES4142000082 41420000 Wholesale trade 4480134 CES4142300057 41423000 Durable goods 3449034 CES4142300082 41423000 Durable goods 2261006 CES4142340057 41423400 Commercial equipment 915034 CES4142400057 41424000 Nondurable goods 2042100 CES4142400082 41424000 Nondurable goods 1367413 CES4142500057 41425000 Electronic markets and agents and brokers 1359605 CES4142500082 41425000 Electronic markets and agents and brokers 855291 CES4142512057 41425120 Wholesale trade agents and brokers 1317589 CES4142512082 41425120 Wholesale trade agents and brokers 831319 CES4200000057 42000000 Retail trade 8857570 CES4200000082 42000000 Retail trade 6123554 CES4244100057 42441000 Motor vehicle and parts dealers 1740340 CES4244100082 42441000 Motor vehicle and parts dealers 1180297 CES4244110057 42441100 Automobile dealers 1232818 CES4244110082 42441100 Automobile dealers 839595 CES4244111057 42441110 New car dealers 1110063 CES4244500057 42445000 Food and beverage stores 1352705 CES4244500082 42445000 Food and beverage stores 974710 CES4244510057 42445100 Grocery stores 1180650 CES4244510082 42445100 Grocery stores 868430 CES4244511057 42445110 Supermarkets and other grocery stores 1120814 CES4244511082 42445110 Supermarkets and other grocery stores 821282 CES4245200057 42452000 General merchandise stores 1437264 CES4245200082 42452000 General merchandise stores 1065807 CES4300000057 43000000 Transportation and warehousing 4588906 CES4300000082 43000000 Transportation and warehousing 3538683 CES4348400057 43484000 Truck transportat

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">Create a Union table comparing average weekly earnings&nbsp; of production and nonsupervisory employees between annual_16 and january_17 using the data type 30.&nbsp; Round to the nearest penny.&nbsp; You should have a column for the average earnings and a column for the year, and the period.</span>

In [174]:
select round(avg(value),2) Avg,period, year
from LaborStatisticsDB.dbo.annual_2016 a,LaborStatisticsDB.dbo.series s,LaborStatisticsDB.dbo.datatype d
where a.series_id=s.series_id AND 
     s.data_type_code=d.data_type_code and 
    d.data_type_code='30' 
    and lower(series_title) ='average weekly earnings of production and nonsupervisory employees'
group by period ,year



UNION
select round(avg(value),2) avg,period, year
from LaborStatisticsDB.dbo.january_2017 j,LaborStatisticsDB.dbo.series s1,LaborStatisticsDB.dbo.datatype d1
where j.series_id=s1.series_id AND 
    s1.data_type_code=d1.data_type_code and 
     d1.data_type_code='30' 
   and lower(series_title) ='average weekly earnings of production and nonsupervisory employees'
group by period,year

      


(2 rows affected)

Total execution time: 00:00:00.188

Avg period year 797.2 M13 2016 808.53 M01 2017

## <span style="font-size:12pt;font-family:Montserrat, sans-serif;color:rgb(53, 74, 95);background-color:rgb(255, 255, 255);font-weight:700;vertical-align:baseline;white-space:pre-wrap;">Summarize Your Results</span>

<span style="font-size:10pt;font-family:Arvo, serif;color:rgb(102, 102, 102);background-color:rgb(255, 255, 255);vertical-align:baseline;white-space:pre-wrap;">With what you know now about the </span> <span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:&quot;Times New Roman&quot;;font-size:13.3333px;white-space:pre-wrap;">Bureau of Labor Statistics (BLS) Current Employment Survey (CES) results and working with the</span> <span style="color:rgb(102, 102, 102);white-space:pre-wrap;font-family:&quot;Times New Roman&quot;;font-size:13.3333px;">Labor Statistics Database</span><span style="background-color:rgb(255, 255, 255);color:rgb(102, 102, 102);font-family:Arvo, serif;font-size:10pt;white-space:pre-wrap;">, answer the following questions. Note that while this is subjective, you should include relevant data to back up your opinion.</span>

1. <span style="background-color:rgb(255, 255, 255);font-size:10pt;white-space:pre-wrap;color:rgb(102, 102, 102);font-family:Arvo, serif;">During which time period did production and nonsupervisory employees fare better?</span>

Double click here to add your answer

<span style="color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 13.3333px; white-space: pre-wrap; background-color: rgb(255, 255, 255);">Ans:M01</span>

<span style="color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 13.3333px; white-space: pre-wrap; background-color: rgb(255, 255, 255);">The avg earning for employees is more (808.53) in the period <b>M01 (month of January)</b> from the data in January_2017 table compared to the data from annual_2016 (highest of 797.2.)</span>

2. <span style="background-color: rgb(255, 255, 255); font-size: 10pt; white-space: pre-wrap; color: rgb(102, 102, 102); font-family: Arvo, serif;">In which industries did production and nonsupervisory employees fare better?</span>

Double click here to add your answer.

  

Ans: Total private

 From the data from <span style="background-color: rgb(255, 255, 255); color: rgb(102, 102, 102); font-family: Arvo, serif; font-size: 13.3333px; white-space: pre-wrap;">January 2021 table, Total private industry has the total weekly payroll the most . We can say that production and nonsupervisory employees in <b>Total private</b> industry fare better.</span>

3. <span style="font-size: 10pt; font-family: Arvo, serif; color: rgb(102, 102, 102); background-color: rgb(255, 255, 255); font-variant-numeric: normal; font-variant-east-asian: normal; vertical-align: baseline; white-space: pre-wrap;">Now that you have explored the datasets, is there any data or information that you wish you had in this analysis?</span>

Double click here to add your answer.

 I think the comparison we did is with 2016 **annual** and **January** 2017. The comparison between 2016 Januay and 2017  January or annual 2016 and annual 2017 would have been given us better results. They should have given us monthly datas of 2016 or annual data of 2017.

Also if they had given average weekly payroll for production and nonsupervisor employees , or number of employees,it would have been better to find the better fare. We have only aggregate now.